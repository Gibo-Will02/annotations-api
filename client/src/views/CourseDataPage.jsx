import React, { useEffect, useState } from 'react';
import styled from 'styled-components';
import axios from 'axios';

function CourseDataPage() {
	const [data,setData] = useState({});
    const [userId, setUserId] = useState("");

	const [updated, setUpdated] = useState(userId);

	const handleChange = (event) => {
		setUserId(event.target.value);
	};

	const handleClick = () => {
		setUpdated(prevUserId => prevUserId !== userId ? userId : prevUserId);
	};

	useEffect(() => {
        if (userId !== "") {
            axios.post('/api/course_data', {'_CID': userId})
			.then(response => {
				setData({...response.data});
				console.log(response.data);
			})
			.catch(error => {
				console.error(error);
			});
        }
	}, [updated]); //[] updates page if value changes, if empty it only updates on entry to the page
	
	/*client/src/views/CourseDataPage.jsx*/
	return (
		<div>
			<label> Enter the course id:
				<input placeholder="Course Id Here" type='text' onChange={handleChange}/>
			</label>
			<button onClick={handleClick}>Search Course</button>
			<h1>Perusall API Course Return:</h1>
			<h2>Course Name: {data.name}</h2> 
			<li>ID: {data._id}</li>
			{(data.instructorIds && data.instructorIds.length > 0) ? (
				data.instructorIds.map(id => <li key={id}>Instructor Id: {id}</li>)
			) : (
				<li>No instructor IDs available</li>
			)}
			{(data.studentIds && data.studentIds.length > 0) ? (
				data.studentIds.map(id => <li key={id}>Student Id: {id}</li>)
			) : (
				<li>No student IDs available</li>
			)}
		</div>
    )
}

export default CourseDataPage;