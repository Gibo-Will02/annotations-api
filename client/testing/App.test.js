
import { render, screen, fireEvent } from '@testing-library/react';
//import App from '../src/App';
import renderer from "react-test-renderer";
import { JSX } from 'react/jsx-dev-runtime';
//const CourseDataPage = require('../src/views/CourseDataPage');
import CourseDataPage from '../src/views/CourseDataPage';
//import InstitutionCoursesPage from '../src/views/InstitutionCoursesPage';
//import InstitutionRosterPage from '../src/views/InstitutionRosterPage';
//import UserCourseListPage from '../src/views/UserCourseListPage';


//#region course data page tests
describe("Jest Snapshot testing suite", () => {
  it("Matches DOM Snapshot", () => {
    const domTree = renderer.create(<CourseDataPage />).toJSON();
    expect(domTree).toMatchSnapshot();
  });
});


test('course data page displays correct information', () => {
  const domTree = renderer.create(<CourseDataPage />).toJSON();
  expect(domTree.includes("Perusall API Course Return:"));
 

});

test('course data page displays correct information', () => {
  render(<CourseDataPage />);
  const TextElement = screen.getByText("Course Name:");
  expect(TextElement).toBeInTheDocument();
 

});

test('course data page has search button and activates', ()=> {
    render(<CourseDataPage />);
    const nButton1 = screen.getByText('Search Course');
    fireEvent.click(nButton1);
    expect(nButton1.toHaveBeenCalled);
});

describe('Api Testing using Fake Data', () => {
  beforeEach(() => {
    fetchMock.resetMocks();
  })

  test('renders info when API call succeeds', async ()=> {
    const fakeUsers = [
      {_id: 1 , instructorIds: 'Jack'},
      {_id: 2 , studentIds: 'Noah'},
    ];
    fetchMock.mockResolvedValue({ status: 200, json: jest.fn(() => fakeUsers) });

    render(<CourseDataPage />);
    const input = screen.getByPlaceholderText('Enter the course id:');
    const value = 'blank';
    fireEvent.change(input, {
      target: {value}
    });
    const nButton1 = screen.getByText('Search Course');
    fireEvent.click(nButton1);

    expect(await screen.findByText('Jack')).toBeInTheDocument();
    expect(await screen.findByText('Noah')).toBeInTheDocument();
  });

  
  
  test('renders failstate when no person is found', async ()=> {
    
    render(<CourseDataPage />);//replace
    const input = screen.getByPlaceholderText('Enter the course id:');
    const value = 'blank';
    fireEvent.change(input, {
      target: {value}
    });
    const nButton1 = screen.getByText('Search Course');
    fireEvent.click(nButton1);

    expect(await screen.findByText('No instructor IDs available')).toBeInTheDocument();
    expect(await screen.findByText('No student IDs available')).toBeInTheDocument();
  });
  
});
//#endregion


