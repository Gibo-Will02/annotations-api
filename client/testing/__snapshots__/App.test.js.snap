// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Api sends information through renders info when API call succeeds 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        style="display: flex; flex-direction: column; align-items: center;"
      >
        <h1>
          Perusall API User Return:
        </h1>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      style="display: flex; flex-direction: column; align-items: center;"
    >
      <h1>
        Perusall API User Return:
      </h1>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Jest Snapshot testing suite Matches AssignmentAnnotationsPage Snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        style="display: flex; flex-direction: column; align-items: center;"
      >
        <div>
          <label>
             Enter the course id:
            <input
              placeholder="Course Id Here"
              type="text"
            />
          </label>
          <label>
             Enter the assignment id:
            <input
              placeholder="Assignment Id Here"
              type="text"
            />
          </label>
          <button>
            Search Assignment
          </button>
        </div>
        <h1
          style="border-bottom: 1px solid black; display: inline-block;"
        >
          Assignment Annotations:
        </h1>
        <label>
          No assignments to display
        </label>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      style="display: flex; flex-direction: column; align-items: center;"
    >
      <div>
        <label>
           Enter the course id:
          <input
            placeholder="Course Id Here"
            type="text"
          />
        </label>
        <label>
           Enter the assignment id:
          <input
            placeholder="Assignment Id Here"
            type="text"
          />
        </label>
        <button>
          Search Assignment
        </button>
      </div>
      <h1
        style="border-bottom: 1px solid black; display: inline-block;"
      >
        Assignment Annotations:
      </h1>
      <label>
        No assignments to display
      </label>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Jest Snapshot testing suite Matches AssignmentGradesPage Snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        style="display: flex; flex-direction: column; align-items: center;"
      >
        <div>
          <label>
             Enter the course id:
            <input
              placeholder="Course Id Here"
              type="text"
            />
          </label>
          <label>
             Enter the assignment id:
            <input
              placeholder="Assignment Id Here"
              type="text"
            />
          </label>
          <button>
            Search Assignment
          </button>
        </div>
        <h1
          style="border-bottom: 1px solid black; display: inline-block;"
        >
          Assignment Grades:
        </h1>
        <label>
          No assignments to display
        </label>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      style="display: flex; flex-direction: column; align-items: center;"
    >
      <div>
        <label>
           Enter the course id:
          <input
            placeholder="Course Id Here"
            type="text"
          />
        </label>
        <label>
           Enter the assignment id:
          <input
            placeholder="Assignment Id Here"
            type="text"
          />
        </label>
        <button>
          Search Assignment
        </button>
      </div>
      <h1
        style="border-bottom: 1px solid black; display: inline-block;"
      >
        Assignment Grades:
      </h1>
      <label>
        No assignments to display
      </label>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Jest Snapshot testing suite Matches AssignmentReportsPage Snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        style="display: flex; flex-direction: column; align-items: center;"
      >
        <div>
          <label>
             Enter the course id:
            <input
              placeholder="Course Id Here"
              type="text"
            />
          </label>
          <label>
             Enter the assignment id:
            <input
              placeholder="Assignment Id Here"
              type="text"
            />
          </label>
          <button>
            Search Assignment
          </button>
        </div>
        <label>
           Select the report type:
          <div
            class="dropdown"
            data-testid="dropTest"
          >
            <button
              aria-expanded="false"
              class="dropdown-toggle btn btn-primary"
              id="reportTypeDropdown"
              type="button"
            >
              Report Type
            </button>
          </div>
        </label>
        <label>
          Current report type: 
          submissionTime
        </label>
        <h1
          style="border-bottom: 1px solid black; display: inline-block;"
        >
          Assignment Annotations:
        </h1>
        <label>
          No assignments to display
        </label>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      style="display: flex; flex-direction: column; align-items: center;"
    >
      <div>
        <label>
           Enter the course id:
          <input
            placeholder="Course Id Here"
            type="text"
          />
        </label>
        <label>
           Enter the assignment id:
          <input
            placeholder="Assignment Id Here"
            type="text"
          />
        </label>
        <button>
          Search Assignment
        </button>
      </div>
      <label>
         Select the report type:
        <div
          class="dropdown"
          data-testid="dropTest"
        >
          <button
            aria-expanded="false"
            class="dropdown-toggle btn btn-primary"
            id="reportTypeDropdown"
            type="button"
          >
            Report Type
          </button>
        </div>
      </label>
      <label>
        Current report type: 
        submissionTime
      </label>
      <h1
        style="border-bottom: 1px solid black; display: inline-block;"
      >
        Assignment Annotations:
      </h1>
      <label>
        No assignments to display
      </label>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Jest Snapshot testing suite Matches CourseAssignmentPage Snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        style="display: flex; flex-direction: column; align-items: center;"
      >
        <div>
          <label>
             Enter the course id:
            <input
              placeholder="Course Id Here"
              type="text"
            />
          </label>
          <button>
            Search Course
          </button>
        </div>
        <h1
          style="border-bottom: 1px solid black; display: inline-block;"
        >
          Assignments (From Perusall):
        </h1>
        <label>
          No assignments to display
        </label>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      style="display: flex; flex-direction: column; align-items: center;"
    >
      <div>
        <label>
           Enter the course id:
          <input
            placeholder="Course Id Here"
            type="text"
          />
        </label>
        <button>
          Search Course
        </button>
      </div>
      <h1
        style="border-bottom: 1px solid black; display: inline-block;"
      >
        Assignments (From Perusall):
      </h1>
      <label>
        No assignments to display
      </label>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Jest Snapshot testing suite Matches DOM Snapshot 1`] = `
<div
  style={
    Object {
      "alignItems": "center",
      "display": "flex",
      "flexDirection": "column",
    }
  }
>
  <label>
     Enter the course id:
    <input
      onChange={[Function]}
      placeholder="Course Id Here"
      type="text"
    />
  </label>
  <button
    onClick={[Function]}
  >
    Search Course
  </button>
  <h1>
    Perusall API Course Return:
  </h1>
  <h2>
    Course Name: 
  </h2>
  <li>
    ID: 
  </li>
  <li>
    No instructor IDs available
  </li>
  <li>
    No student IDs available
  </li>
</div>
`;

exports[`Jest Snapshot testing suite Matches DOM Snapshot 2`] = `
<div
  style={
    Object {
      "alignItems": "center",
      "display": "flex",
      "flexDirection": "column",
    }
  }
>
  <div>
    <label>
      Loading data
    </label>
  </div>
</div>
`;
